test/functional/page_objects/discover_page.ts

```diff
@@ -185,6 +191,46 @@ export function DiscoverPageProvider({ getService, getPageObjects }: FtrProvider
       return await docHeader.getVisibleText();
     }
 
+    public async getDocTableRows() {
+      await header.waitUntilLoadingHasFinished();
+      const rows = await testSubjects.findAll('docTableRow');
+      return rows;
+    }
+
+    /**
+     * Expanding a row has the side effect to increase the number of raw rows in the table.
+     * So be careful to manipulate the DOM here.
+     * Use getDocTableRows() to have the clean list of rows.
+     */
+    public async expandToggleDocTableRow(index: number) {
+      const rows = await this.getDocTableRows();
+      // The index above was meant to work with CSS nth-child, but here we deal with JS arrays :)
+      const indexWithOffset = Math.max(0, index - 1);
+      const toggleEl = await testSubjects.findDescendant(
+        `docTableExpandToggleColumn`,
+        rows[indexWithOffset]
+      );
+      await toggleEl.click();
+      await header.waitUntilLoadingHasFinished();
```

Comment:
Could you check if this function is a duplicate of 

https://github.com/elastic/kibana/blob/2399780d99a812a3d9a0c1957ea47582e0107072/test/functional/services/doc_table.ts#L68

I think this was for created mainly for Discover context (`isAnchorRow`), but since Discover Main and Context using the same doc table component it looks like a duplicate. many thx!